API/SearchService/src/util.py

```diff
@@ -0,0 +1,74 @@
+from enum import Enum
+import azure.functions as func
+
+from .exception import ParameterException
+
+class SearchType(int, Enum):
+    All = 1
+    Scenario = 2
+    Command = 3
+
+    def get_search_fields(self):
+        if self == self.All:
+            return ["scenario", "description", "commandSet/command"]
+        elif self == self.Scenario:
+            return ["scenario", "description"]
+        elif self == self.Command:
+            return ["commandSet/command"]
+
+
+class RequiredParameter:
+    pass
+
+
+def get_param(req: func.HttpRequest, name: str, default=RequiredParameter):
+    value = req.params.get(name)
+    if not value:
+        try:
+            req_body = req.get_json()
+        except ValueError:
+            pass
+        else:
+            value = req_body.get(name, default)
+    if value == RequiredParameter:
+        raise ParameterException(f'Illegal parameter: please pass in the parameter "{name}"')
```

Comment:
The judgment logic is not intuitive enough. It is recommended to use `is_required` parameter to indicate whether this parameter is required. Then the `ParameterException` will be thrown when the judgment condition is `if not value and is_required:`